share/bin/harvest.py

```diff
@@ -0,0 +1,136 @@
+import re
+import os
+
+import pendulum
+
+from share import tasks
+from share.bin.util import command
+from share.harvest.scheduler import HarvestScheduler
+from share.models import SourceConfig
+
+
+@command('Fetch data to disk or stdout, using the specified SourceConfig')
+def fetch(args, argv):
+    """
+    Usage: {0} fetch <sourceconfig> [<date> | --start=YYYY-MM-DD --end=YYYY-MM-DD] [--limit=LIMIT] [--print | --out=DIR] [--set-spec=SET]
+
+    Options:
+        -l, --limit=NUMBER      Limit the harvester to NUMBER of documents
+        -p, --print             Print results to stdout rather than to a file
+        -o, --out=DIR           The directory to store the fetched data in. Defaults to ./fetched/<sourceconfig>
+        -s, --start=YYYY-MM-DD  The date at which to start fetching data.
+        -e, --end=YYYY-MM-DD    The date at which to stop fetching data.
+        -i, --ignore-disabled   Allow disabled SourceConfigs to run.
+        --set-spec=SET          The OAI setSpec to limit harvesting to.
+    """
+
+    try:
+        config = SourceConfig.objects.get(label=(args['<sourceconfig>']))
+    except SourceConfig.DoesNotExist:
+        print('SourceConfig "{}" not found.'.format(args['<sourceconfig>']))
+        return -1
+
+    harvester = config.get_harvester(pretty=True)
+
+    kwargs = {k: v for k, v in {
+        'limit': int(args['--limit']) if args.get('--limit') else None,
+        'set_spec': args.get('--set-spec'),
+        'ignore_disabled': args.get('ignore_disabled'),
+    }.items() if v is not None}
+
+    if not args['<date>'] and not (args['--start'] and args['--end']):
+        gen = harvester.fetch(**kwargs)
+    elif args['<date>']:
+        gen = harvester.fetch_date(pendulum.parse(args['<date>']), **kwargs)
+    else:
+        gen = harvester.fetch_date_range(pendulum.parse(args['--start']), pendulum.parse(args['--end']), **kwargs)
+
+    if not args['--print']:
+        args['--out'] = args['--out'] or os.path.join(os.curdir, 'fetched', config.label)
+        os.makedirs(args['--out'], exist_ok=True)
+
+    for result in gen:
+        if args['--print']:
+            print('Harvested data with identifier "{}"'.format(result.identifier))
+            print(result.datum)
+            print('\n')
+        else:
+            suffix = '.xml' if result.datum.startswith('<') else '.json'
+            with open(os.path.join(args['--out'], re.sub(r'[:\\\/\?\*]', '', str(result.identifier))) + suffix, 'w') as fobj:
+                fobj.write(result.datum)
+
+
+@command('Harvest data using the specified SourceConfig')
+def harvest(args, argv):
+    """
+    Usage:
+        {0} harvest <sourceconfig> [<date>] [-nflj] [options]
+        {0} harvest <sourceconfig> [<date>] [-afsj] [options]
+        {0} harvest <sourceconfig> --all [<date>] [-afsj | -nflj] [options]
+        {0} harvest <sourceconfig> (--start=YYYY-MM-DD> --end=YYYY-MM-DD) [-afsj | -nflj] [options]
```

Comment:
For some reason `args['--start']` contains a list (e.g. `['2017-05-01']`), which kills pendulum below, while `args['--end']` correctly has a string. My best guess was the `>` here, but removing it didn't seem to fix anything, so I'm lost.